Lib9c/Action/ClaimItems.cs

```diff
@@ -0,0 +1,113 @@
+            IImmutableDictionary<string, IValue> plainValue)
+        {
+            AvatarAddresses = ((List)plainValue[AvatarAddressKey]).Select(x => x.ToAddress());
+            Amounts = ((List)plainValue[AmountKey]).Select(x => x.ToFungibleAssetValue());
+        }
+
+        public override IAccount Execute(IActionContext context)
+        {
+            context.UseGas(1);
+
+            var states = context.PreviousState;
+            var itemSheet = states.GetSheets(containItemSheet: true).GetItemSheet();
+
+            var inventories = AvatarAddresses.Select(address =>
+            {
+                var addressHex = GetSignerAndOtherAddressesHex(context, address);
+                var inventory = states.GetInventory(address.Derive(LegacyInventoryKey));
+                if (inventory is null)
+                {
+                    throw new FailedLoadStateException(ActionTypeText, addressHex, typeof(AvatarState), address);
+                }
+
+                return inventory;
+            })
+                .ToList();
+
+            foreach (var fungibleAssetValue in Amounts)
+            {
+                var ticker = fungibleAssetValue.Currency.Ticker;
+                if (!ticker.StartsWith("IT_") ||
+                    !int.TryParse(ticker.Replace("IT_", string.Empty), out var itemId))
+                {
+                    throw new ArgumentException($"Format of Amount currency's ticker is invalid");
+                }
+
+                var decimalPlaces = fungibleAssetValue.Currency.DecimalPlaces;
+                if (decimalPlaces != 0)
+                {
+                    throw new ArgumentException(
+                        "DecimalPlaces of fungibleAssetValue for claimItems are not 0");
+                }
+
+                var balance = states.GetBalance(context.Signer, fungibleAssetValue.Currency);
+                if (balance < fungibleAssetValue)
+                {
+                    throw new NotEnoughFungibleAssetValueException(
+                        context.Signer.ToHex(),
+                        fungibleAssetValue.RawValue,
+                        balance);
+                }
+
+                var item = itemSheet[itemId] switch
+                {
+                    MaterialItemSheet.Row materialRow =>
+                        ItemFactory.CreateTradableMaterial(materialRow),
+                    var itemRow => ItemFactory.CreateItem(itemRow, context.Random)
+                };
+
+                foreach (var inventory in inventories)
+                {
+                    inventory.AddItem(item, (int)fungibleAssetValue.RawValue);
+                }
+                states = states.BurnAsset(context, context.Signer, fungibleAssetValue * inventories.Count);
+            }
+
+            foreach (var (avatarAddress, i) in AvatarAddresses.Select((x, i) => (x, i)))
+            {
+                states = states.SetState(avatarAddress.Derive(LegacyInventoryKey), inventories[i].Serialize());
```

Comment:
`avatarAddress.Derive(LegacyInventoryKey)` seems used twice. How about caching the addresses or making a function to derive inventory address in same way.